### PR TITLE
Gh workflow

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: .github/workflows/build_images.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build the ci image
+        run: docker build . --file Containerfile --target ci-base --tag pulp/pulp-ci:latest
+      - name: Build the single-container image
+        run: docker build . --file Containerfile --tag pulp/pulp-fedora31:latest
+      - name: Greet on gh_workflow
```
Comment (fao89) [inline .github/workflows/build_images.yaml:20]:
```suggestion
      - name: Docker login
        env:
          DOCKER_BOT_PASSWORD: ${{ secrets.DOCKER_BOT_PASSWORD }}
          DOCKER_BOT_USERNAME: ${{ secrets.DOCKER_BOT_USERNAME }}
        run: echo "$DOCKER_BOT_PASSWORD" | docker login -u "$DOCKER_BOT_USERNAME" --password-stdin docker.io
      - name: Greet on gh_workflow
```

File: .github/workflows/build_images.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        if: ${{ branch == "gh_workflow" }}
+        run: echo "On branch gh_workflow; No pushing today!"
+      - name: Push ci image to dockerhub
+        if: ${{ branch == "master" }}
+        run: echo "On branch master; docker push pulp/pulp-ci:latest"
```
Comment (fao89) [inline .github/workflows/build_images.yaml:25]:
```suggestion
        run: |
           docker push docker.io/pulp/pulp-ci:latest
           docker push docker.io/pulp/pulp-fedora31:latest
```

File: .github/workflows/build_images.yaml (hunk 1 above)
Comment (daviddavis) [reply .github/workflows/build_images.yaml:20]: I added DOCKER_BOT_USERNAME and DOCKER_BOT_PASSWORD as secrets in GHA for this repo.

File: .github/workflows/build_images.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build the ci image
+        run: docker build . --file Containerfile --target ci-base --tag pulp/pulp-ci:latest
+      - name: Build the single-container image
+        run: docker build . --file Containerfile --tag pulp/pulp-fedora31:latest
+      - name: Greet on gh_workflow
+        if: ${{ branch == "gh_workflow" }}
```
Comment (fao89) [inline .github/workflows/build_images.yaml:21]:
```suggestion
        if: "contains(github.ref, 'gh_workflow')"
```

File: .github/workflows/build_images.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        run: echo "On branch gh_workflow; No pushing today!"
+      - name: Push ci image to dockerhub
+        if: ${{ branch == "master" }}
```
Comment (fao89) [inline .github/workflows/build_images.yaml:24]:
```suggestion
        if: "contains(github.ref, 'master')"
```

Comment (fao89) [reply .github/workflows/build_images.yaml:24]: https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

File: .github/workflows/build_images.yaml (hunk 1 above)
Comment (fao89) [reply .github/workflows/build_images.yaml:20]: https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets

Comment (mdellweg) [reply .github/workflows/build_images.yaml:20]: Are those secrets limited to being available on master push actions?

File: .github/workflows/publish_images.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+name: Build and publish OCI Images
+
+on:
+  push:
+    branches:
+      - master
+
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply .github/workflows/publish_images.yaml:7]: Maybe, we could also trigger a "manual" rebuild from the release scripts.

[the remark this answers is not in the document]
Comment (daviddavis) [reply .github/workflows/publish_images.yaml:7]: What about if we just add a cron task for now?

```suggestion
  schedule:
    # * is a special character in YAML so you have to quote this string
    - cron: '0 1 * * *'
```

**Pull request merged**